### ISSUE TITLE
Better documentation on Library management.

### ISSUE THREAD

**Issue by xekoukou** [agda/agda#2499]
The [documentation](https://agda.readthedocs.io/en/v2.5.2/tools/package-system.html) is correct but confusing and it has to do with the existence of metavariables.

1.  AGDA_STDLIB is a metavariable while the name of the library is not, it is of a specific library, the standard library.
2. AGDA_STDLIB is used as a metavariable in the documentation. AGDA_DIR is also capitalized but it is an actual variable that depends on the system you are.

All of these make the documentation hard to follow even though configuring agda to use any library is easy. 

edited.

**Comment by UlfNorell** [comment]
(I assume you mean `AGDA_DIR` and not `AGDA_BIN`)

I'm not sure what's confusing about (1). The text is pretty clear about `AGDA_STDLIB` being a metavariable. Is it the fact that the standard library refers to a specific library that's confusing? If so, that could be addressed by linking to where you can download it when it's first mentioned.

`AGDA_DIR` is also a metavariable, used in the same way as `AGDA_STDLIB` to refer to a directory on your system. What may be confusing you is that there's also a `$AGDA_DIR` environment variable which (if set) tells Agda where to find the `AGDA_DIR`. I'm not sure how to address this confusion--referring to the environment variable as `$AGDA_DIR` would perhaps make it clearer, except I believe environment variables have different syntax on Windows.

**Comment by xekoukou** [comment]
I tried to make a guess on what confused me a few weeks ago. I don't know for sure what actually confused me.

**Comment by xekoukou** [comment]
For example, the fact that the standard-library.agda-lib should exist in the library folder is not necessary.

**Comment by UlfNorell** [comment]
I'm not sure what you mean. The [standard library](https://github.com/agda/agda-stdlib) *does* have `standard-library.agda-lib` in it's root directory, and any library you want to use needs to have a `.agda-lib` file.

**Comment by xekoukou** [comment]
Yes, but the users of the library do not need to know that. The standard library developers need to know that.

**Comment by UlfNorell** [comment]
I think it's helpful for people reading the documentation to know that the thing you put in your `libraries` file refers to an actual file. Keep in mind that the example isn't (only) there to teach people how to use the standard library, but how to use any library, including libraries they create themselves.

**Comment by xekoukou** [comment]
That is why I made point 1. You can't do everything in one paragraph.

First paragraph : The min info to use the standard library.
Second paragraph : The minimum info to use any library.
Second section : How to write your own library.

**Comment by banacorn** [comment]
The documentation is really confusing. 

It takes a lot of effort to keep metavariables like `AGDA_STDLIB` and `AGDA_DIR` in mind while trying to figure things out.

**Comment by UlfNorell** [comment]
There has been some improvements to the documentation since this issue was filed, so closing. Suggestions for further improvements are of course always welcome.
